tests/components/automation/test_init.py

```diff
@@ -572,3 +567,50 @@ def test_reload_config_handles_load_fails(self):
         self.hass.bus.fire('test_event')
         self.hass.block_till_done()
         assert len(self.calls) == 2
+
+
+@asyncio.coroutine
+def test_automation_restore_state(hass):
+    """Ensure states are restored on startup."""
+    time = dt_util.utcnow()
+
+    mock_restore_cache(hass, (
+        State('automation.hello', STATE_ON),
+        State('automation.bye', STATE_OFF, {'last_triggered': time}),
+    ))
+
+    config = {automation.DOMAIN: [{
+        'alias': 'hello',
+        'trigger': {
+            'platform': 'event',
+            'event_type': 'test_event_hello',
+        },
+        'action': {'service': 'test.automation'}
+    }, {
+        'alias': 'bye',
+        'trigger': {
+            'platform': 'event',
+            'event_type': 'test_event_bye',
+        },
+        'action': {'service': 'test.automation'}
+    }]}
+
+    assert (yield from async_setup_component(hass, automation.DOMAIN, config))
+
+    state = hass.states.get('automation.hello')
+    assert state
+    assert state.state == STATE_ON
+
+    calls = mock_service(hass, 'test', 'automation')
+
+    state = hass.states.get('automation.bye')
+    assert state
+    assert state.state == STATE_OFF
+    assert state.attributes.get('last_triggered') == time
+    hass.bus.fire('test_event_bye')
+    yield from hass.async_block_till_done()
+    assert len(calls) == 0
+
+    hass.bus.fire('test_event_hello')
+    yield from hass.async_block_till_done()
+    assert len(calls) == 1
```

Comment:
This seems to fail for py3.5 & 3.6, but always passes on 3.4

@pvizeli any ideas?